docs/Basics-Network.md

```diff
@@ -35,121 +37,47 @@ fetch('https://mywebsite.com/endpoint/', {
-        // Handle error
+  getMoviesFromApiAsync() {
+    return fetch('http://facebook.github.io/react-native/movies.json')
+      .then((response) => response.json())
+      .then((responseJson) => {
+        return responseJson.movies;
+      })
+      .catch((error) => {
         console.error(error);
-      }
-    }
-    ...
+      });
   }
   ```
 
-- Note: Errors thrown by rejected Promises need to be caught, or they will be swallowed silently
-
-## WebSocket
-
-WebSocket is a protocol providing full-duplex communication channels over a single TCP connection.
-
-```js
-var ws = new WebSocket('ws://host.com/path');
-
-ws.onopen = () => {
-  // connection opened
-  ws.send('something');
-};
-
-ws.onmessage = (e) => {
-  // a message was received
-  console.log(e.data);
-};
-
-ws.onerror = (e) => {
-  // an error occurred
-  console.log(e.message);
-};
-
-ws.onclose = (e) => {
-  // connection closed
-  console.log(e.code, e.reason);
-};
-```
-
-## XMLHttpRequest
-
-XMLHttpRequest API is implemented on-top of [iOS networking apis](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/URLLoadingSystem/URLLoadingSystem.html) and [OkHttp](http://square.github.io/okhttp/). The notable difference from web is the security model: you can read from arbitrary websites on the internet since there is no concept of [CORS](http://en.wikipedia.org/wiki/Cross-origin_resource_sharing).
+You can also use ES7's `async`/`await` syntax in React Native app:
 
-```js
-var request = new XMLHttpRequest();
-request.onreadystatechange = (e) => {
-  if (request.readyState !== 4) {
-    return;
-  }
-
-  if (request.status === 200) {
-    console.log('success', request.responseText);
-  } else {
-    console.warn('error');
+  ```js
+  async getMoviesFromApi() {
+    try {
+      let response = await fetch('http://facebook.github.io/react-native/movies.json');
+      let responseJson = await response.json();
+      return responseJson.movies;
+    } catch(error) {
+      console.error(error);
+    }
   }
-};
-
-request.open('GET', 'https://mywebsite.com/endpoint.php');
-request.send();
-```
+  ```
 
-You can also use -
+Don't forget to catch any errors that may be thrown by `fetch`, otherwise they will be dropped silently.
 
-```js
-var request = new XMLHttpRequest();
-
-function onLoad() {
-    console.log(request.status);
-    console.log(request.responseText);
-};
-
-function onTimeout() {
-    console.log('Timeout');
-    console.log(request.responseText);
-};
-
-function onError() {
-    console.log('General network error');
-    console.log(request.responseText);
-};
-
-request.onload = onLoad;
-request.ontimeout = onTimeout;
-request.onerror = onError;
-request.open('GET', 'https://mywebsite.com/endpoint.php');
-request.send();
-```
+## WebSocket Support
 
+React Native supports [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket), a protocol which provides full-duplex communication channels over a single TCP connection.
```

Comment:
can we have WebSocket somewhere within our own autogenerated docs? Then you could link to that. I'm suspicious that the mozilla.org link defines a 95%-the-same-but-5%-different thing.